alias/gui.c

```diff
@@ -53,7 +53,31 @@ int alias_sort_name(const void *a, const void *b)
   const struct AliasView *av_a = a;
   const struct AliasView *av_b = b;
 
-  int r = mutt_str_coll(av_a->alias->name, av_b->alias->name);
+  int r;
+
+  if (av_a->is_visible == av_b->is_visible) {
+    r = mutt_str_coll(av_a->alias->name, av_b->alias->name);
+  } else {
+    r = (av_a->is_visible == av_b->is_visible) ? 0 : av_a->is_visible ? -1 : 1;
+  }
+
+  return RSORT(r);
```

Comment:
I think we need to deal with the `is_visible` case first, and `return`.
The code above applies `RSORT(r)` to the result, even if hidden.
If `sort = reverse-name`, I _think_ we might get:
```
hidden1
hidden2
zebra
yak
xebu
```

Reply:
Most likely I am misunderstanding something.

How would you """intersect""" two different values in a sorting function?
If I deal with the `is_visible` case first, what about the `name` sorting?

Reply:
If only one is visible, then `$sort` doesn't matter.
Make sure the hidden Alias is moved towards the end.

```c
if (av_a->is_visible != av_b->is_visible)
  return av_a->is_visible ? -1 : 1;
```

Check for both being invisible:

```c
if (!av_a->is_visible)
  return 0;
```

Then sort by name (honouring `reverse-`):

```c
int r = mutt_str_coll(av_a->alias->name, av_b->alias->name);
return RSORT(r);
```

(I haven't tested this)

Reply:
I _think_ I got it working... but I will apply the same logic to the other sorting functions.
I still have to apply all these changes to the query menu, which is something that I will do tomorrow. :-)